### ISSUE TITLE
Clear button does not set null value

### ISSUE THREAD

**Issue by yuriy-fix** [vaadin/vaadin-combo-box-flow#277]
@mpetzold commented on [Mon Jul 01 2019](https://github.com/vaadin/vaadin-combo-box/issues/823)

I have a ComboBox with values. When I select a value everything is fine. However, when I click the clear button, the originally selected value is set instead of null. If I then click somewhere outside of the ComboBox and click the clear button again, a null value is set.

Can be reproduced using a ValueChangeListener:

```
combobox.addValueChangeListener(new ValueChangeListener<ComponentValueChangeEvent<ComboBox<MyObject>, MyObject>>() {
				@Override
				public void valueChanged(ComponentValueChangeEvent<ComboBox<MyObject>, MyObject> event) {
					System.out.println(event.getValue());
				}
				
			});
```

Vaadin / Flow version: 14.0.0.rc2
Java version: 1.8
OS version: Debian 9
Browser: Chrome 75.0.3770.100

**Comment by yuriy-fix** [comment]
Cannot reproduce the issue. @mpetzold, could you please provide more details on this issue (i.e the initialisation of the comboBox, bean type used) ?

**Comment by mpetzold** [comment]
`
ComboBox<Organization> parent = new ComboBox<Organization>("Organization");
parent.setClearButtonVisible(true);
parent.setItems(...);
parent.setValue(...);
`

**Comment by mpetzold** [comment]
IMHO: This is a bug and critical (maybe frontend componente). You should clarify if empty string or null is the default behavior. Otherwise there will be a lot of mess out there!

I also realized that text fields now have empty string instead of null as default value. A breaking change between Vaadin 8 and Vaadin 10+. Maybe this is related.

**Comment by pekam** [comment]
The issue has been fixed in the web component: https://github.com/vaadin/vaadin-combo-box/pull/824

Test case for the Flow part was added in #281.

The fix will be included in Vaadin 14.0.0.rc4.

> You should clarify if empty string or null is the default behavior. Otherwise there will be a lot of mess out there!

`ComboBox` has a generic type parameter for its items. It can contain e.g. `Person` data objects instead of strings. This is why the empty value can't be an empty string. Also, having `null` as the empty value makes more sense for a component which allows you to select from a pre-defined set of options. I.e. "no selection" is best represented as `null` in Java.

> I also realized that text fields now have empty string instead of null as default value. A breaking change between Vaadin 8 and Vaadin 10+. Maybe this is related.

For `TextField`, it makes more sense to always contain a string value. I'm actually quite surprised to hear that it used to have `null` value in Vaadin 8.
